### PR TITLE
Add PInvoke native tests under Desktop

### PR DESCRIPTION
**Pull request by jiangzeng**

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
When git indicates the file is binary but isn't, this is usually because the text encoding is off. You can fix this by re-saving the file using notepad.exe and setting the encoding to UTF8 or ASCII.

File: tests/src/Interop/Desktop/PInvokeTest/Attributes/LCID/LCIDNative.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+{
+    size_t lenstrFalseReturn = strlen(strFalseReturn);
+    LPSTR ret = (LPSTR)(CoTaskMemAlloc(sizeof(char)*(lenstrFalseReturn+1)));
+    memset(ret,'\0',lenstrFalseReturn+1);
+    strncpy_s(ret,lenstrFalseReturn + 1,strFalseReturn,1);
+    return ret;
+}
+
+//Test Method2
+extern "C" __declspec(dllexport) LPSTR Marshal_InOut1(int lcid, LPSTR s)
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/Desktop/PInvokeTest/Attributes/LCID/LCIDNative.cpp]: Look for `DLLEXPORT` and you should include `<xplatform.h>` - see existing tests.

Comment (jiangzeng) [reply tests/src/Interop/Desktop/PInvokeTest/Attributes/LCID/LCIDNative.cpp]: Thanks, should be fixed

File: tests/src/Interop/Desktop/PInvokeTest/Attributes/LCID/LCIDNative.cpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    //In-Place Change
+    strncpy_s(s,len + 1,strNative,lenstrNative);
+
+    //Return
+    return ReturnString();
+}
+
+extern "C" __declspec(dllexport) LPSTR Marshal_InOut2(LPSTR s,int lcid)
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/Desktop/PInvokeTest/Attributes/LCID/LCIDNative.cpp]: @jiangzeng Did you update these throughout? `__declspec(dllexport)` -> `DLLEXPORT`?

File: tests/src/Interop/Desktop/PInvokeTest/DateTime/NativeDateTime.cpp [hunk 3]
```diff
@@ -0,0 +1,225 @@
+// NativeDateTime.cpp : Defines the exported functions for the DLL application.
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/Desktop/PInvokeTest/DateTime/NativeDateTime.cpp:5]: Remember that _all_ source files must have the license banner at the top - see the `readme.md` in the `Interop/` directory.

File: tests/src/Interop/Desktop/PInvokeTest/DateTime/NativeDateTime.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// The following ifdef block is the standard way of creating macros which make exporting 
+// from a DLL simpler. All files within this DLL are compiled with the NATIVEDATETIME_EXPORTS
+// symbol defined on the command line. this symbol should not be defined on any project
+// that uses this DLL. This way any other project whose source files include this file see 
+// NATIVEDATETIME_API functions as being imported from a DLL, whereas this DLL sees symbols
+// defined with this macro as being exported.
+#ifdef NATIVEDATETIME_EXPORTS
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/Desktop/PInvokeTest/DateTime/NativeDateTime.h]: Remove all of these export style `#ifdef`. Instead use the `DLLEXPORT` macro.

File: tests/src/Interop/Desktop/PInvokeTest/Primitives/Int/PInvokeIntTest.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            Assert.AreEqual(intReturn, MarshalPointer_InOut(ref int5), "The return value is wrong");
+            Assert.AreEqual(intNative, int5, "The passed value is wrong");
+
+            int int6 = intManaged;
+            Assert.AreEqual(intReturn, MarshalPointer_Out(out int6), "The return value is wrong");
+            Assert.AreEqual(intNative, int6, "The passed value is wrong");
+
+            return 100;
+        } catch (Exception e){
+            Console.WriteLine("Test failure: " + e.Message);
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/Desktop/PInvokeTest/Primitives/Int/PInvokeIntTest.cs]: Can you please update the `readme.md`. This line should be `Console.WriteLine($"Test failure: {e}");`

File: tests/src/Interop/Desktop/PInvokeTest/Attributes/LCID/LCIDNative.cpp (hunk 2 above)
Comment (jiangzeng) [reply tests/src/Interop/Desktop/PInvokeTest/Attributes/LCID/LCIDNative.cpp]: Just updated all __declspec(dllexport) -> DLL_EXPORT

File: tests/src/Interop/Desktop/PInvokeTest/DateTime/NativeDateTime.cpp (hunk 3 above)
Comment (jiangzeng) [reply tests/src/Interop/Desktop/PInvokeTest/DateTime/NativeDateTime.cpp:5]: Added the license banner for all C# & C++ source files

File: tests/src/Interop/Desktop/PInvokeTest/DateTime/NativeDateTime.h (hunk 4 above)
Comment (jiangzeng) [reply tests/src/Interop/Desktop/PInvokeTest/DateTime/NativeDateTime.h]: Fixed

File: tests/src/Interop/Desktop/PInvokeTest/Primitives/Int/PInvokeIntTest.cs (hunk 5 above)
Comment (jiangzeng) [reply tests/src/Interop/Desktop/PInvokeTest/Primitives/Int/PInvokeIntTest.cs]: Fixed

File: tests/src/Interop/Desktop/PInvokeTest/BestFitMapping/Char/AFF_PFF/AFF_PFF.cs [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+    }
+
+    public static int Main()
+    {
+        if (System.Globalization.CultureInfo.CurrentCulture.Name != "en-US")
+        {
+            Console.WriteLine("Non english platforms are not supported");
+            Console.WriteLine("passing without running tests");
+
+            Console.WriteLine("--- Sucess");
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/Desktop/PInvokeTest/BestFitMapping/Char/AFF_PFF/AFF_PFF.cs]: Spelling 'Success'

**Pull request closed without merging**